File: leetcode/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        pos = 0
        is_equal = True

        while is_equal:

            ch = ''
            for i in range(len(strs)):

                if pos > len(strs[i])-1:
                    is_equal = False
                    break

                if i == 0:
                    ch = strs[i][pos]
                else:
                    if strs[i][pos] != ch:
                        is_equal = False
                        break
            
            if is_equal:
                pos +=1
            else:
                break
        
        return strs[0][:pos]
    


    class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = ''
        for i in range(len(strs[0])):
            for s in strs:
                if i == len(s) or s[i] != strs[0][i]:
                    return res
            res += strs[0][i]
        return res

            
       

File: yandex_back_prep_cpp/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int i = 0;
        vector<char> prefix;

        vector<int> lens;
        for (auto s: strs) {
            lens.push_back(s.size());
        }

        int min_size = *min_element(lens.begin(), lens.end());

        for (int i=0; i<min_size; i++) {
            set<char> c;
            for (auto s: strs){
                c.insert(s[i]);
            }
            if (c.size() == 1) {
                prefix.push_back(*(c.begin()));
            } else {
                break;
            }

        }
 
        return string(prefix.begin(), prefix.end());

    }
};